spring-rabbit/src/main/java/org/springframework/amqp/rabbit/core/RabbitTemplate.java

```diff
@@ -2378,14 +2418,34 @@ public void doSend(Channel channel, String exchangeArg, String routingKeyArg, Me
 			logger.debug("Publishing message [" + messageToUse
 					+ "] on exchange [" + exch + "], routingKey = [" + rKey + "]");
 		}
-		sendToRabbit(channel, exch, rKey, mandatory, messageToUse);
+		observeTheSend(channel, messageToUse, mandatory, exch, rKey);
 		// Check if commit needed
 		if (isChannelLocallyTransacted(channel)) {
 			// Transacted channel created by this template -> commit.
 			RabbitUtils.commitIfNecessary(channel);
 		}
 	}
 
+	protected void observeTheSend(Channel channel, Message message, boolean mandatory, String exch, String rKey) {
+
+		if (!this.observationRegistryObtained) {
+			obtainObservationRegistry(this.applicationContext);
+			this.observationRegistryObtained = true;
+		}
+		Observation observation;
+		ObservationRegistry registry = getObservationRegistry();
+		if (!this.observationEnabled || registry == null) {
+			observation = Observation.NOOP;
+		}
+		else {
+			observation = RabbitTemplateObservation.TEMPLATE_OBSERVATION.observation(this.observationConvention,
+					DefaultRabbitTemplateObservationConvention.INSTANCE,
+						new RabbitMessageSenderContext(message, this.beanName, exch + "/" + rKey), registry);
```

Comment:
If we took a `Supplier<Context>` here instead of `Context` where the supplier was only called if `ObservationRegistry` wasn't a no-op, and therefore the `Context` object wouldn't be allocated in the no-op case, would that simplify things for you? I think then you wouldn't need the if-else this is contained in, right?

Reply:
+1 from me as well. And I probably agree with @marcingrzejszczak as well.
Only the problem that @garyrussell is off today and next Monday is release day.
I can fix this quickly myself, but we have to be sure that this one is good to go for merging.

Thank you!

Reply:
Or, we could just use a singleton for the No-op case.

Reply:
There is one, Gary, in the `DocumentedObservation`: 
```
    default <T extends Observation.Context> Observation createNotStarted(
            @Nullable ObservationConvention<T> customConvention, @NonNull ObservationConvention<T> defaultConvention,
            @NonNull Supplier<T> contextSupplier, @NonNull ObservationRegistry registry) {
        if (registry.isNoop()) {
            return Observation.NOOP;
        }
        return observation(customConvention, defaultConvention, contextSupplier.get(), registry);
    }
```
I'm not sure why name is like that, but looks like it does exactly what is expected from us here.

Reply:
We're working on unifying the names. Sorry about the commotion cc @ttddyy